app/src/main/java/fr/free/nrw/commons/contributions/ContributionsPresenter.java

```diff
@@ -3,104 +3,155 @@
 import android.content.Context;
 import android.database.Cursor;
 import android.database.DataSetObserver;
-import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Log;
 
-import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
-import androidx.loader.content.CursorLoader;
-import androidx.loader.content.Loader;
+import androidx.lifecycle.LifecycleOwner;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.Observer;
+
+import java.util.ArrayList;
+import java.util.List;
 
 import javax.inject.Inject;
+import javax.inject.Named;
 
+import fr.free.nrw.commons.CommonsApplication;
 import fr.free.nrw.commons.Media;
+import fr.free.nrw.commons.auth.SessionManager;
 import fr.free.nrw.commons.contributions.ContributionsContract.UserActionListener;
+import fr.free.nrw.commons.db.AppDatabase;
+import fr.free.nrw.commons.di.CommonsApplicationModule;
+import fr.free.nrw.commons.mwapi.UserClient;
+import fr.free.nrw.commons.utils.ExecutorUtils;
+import fr.free.nrw.commons.utils.NetworkUtils;
+import io.reactivex.Observable;
+import io.reactivex.Scheduler;
+import io.reactivex.SingleObserver;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
 import timber.log.Timber;
 
-import static fr.free.nrw.commons.contributions.ContributionDao.Table.ALL_FIELDS;
-import static fr.free.nrw.commons.contributions.ContributionsContentProvider.BASE_URI;
-import static fr.free.nrw.commons.settings.Prefs.UPLOADS_SHOWING;
+import static fr.free.nrw.commons.contributions.Contribution.STATE_COMPLETED;
 
 /**
  * The presenter class for Contributions
  */
-public class ContributionsPresenter extends DataSetObserver implements UserActionListener {
+public class ContributionsPresenter implements UserActionListener {
 
     private final ContributionsRepository repository;
+    private final Scheduler mainThreadScheduler;
+    private final Scheduler ioThreadScheduler;
+    private CompositeDisposable compositeDisposable;
     private ContributionsContract.View view;
-    private Cursor cursor;
+    private List<Contribution> contributionList=new ArrayList<>();
 
     @Inject
     Context context;
 
     @Inject
-    ContributionsPresenter(ContributionsRepository repository) {
+    UserClient userClient;
+
+    @Inject
+    AppDatabase appDatabase;
+
+    @Inject
+    SessionManager sessionManager;
+    private LifecycleOwner lifeCycleOwner;
+
+    @Inject
+    ContributionsPresenter(ContributionsRepository repository, @Named(CommonsApplicationModule.MAIN_THREAD) Scheduler mainThreadScheduler,@Named(CommonsApplicationModule.IO_THREAD) Scheduler ioThreadScheduler) {
         this.repository = repository;
+        compositeDisposable=new CompositeDisposable();
```

Comment:
Why is the CompositeDisposable being assigned twice?

Reply:
My bad, removing it